pyelftool_parser/src/analyzer.py

```diff
@@ -0,0 +1,203 @@
+            for i, name in enumerate(sorted_names):
+                if (i == 0):
+                    continue
+                prev_sym = sym_info[sorted_names[i - 1]]
+                cur_sym = sym_info[name]
+                prev_sym['padding'] = cur_sym['address'] - prev_sym['address'] - prev_sym['size']
+                #sort by size
+            sorted_names = sorted(sym_info.keys(),key=lambda name: sym_info[name]['size'], reverse=True)
+            #print symbols in order of size with padding
+            for name in sorted_names[:10]:
+                cur_sym = sym_info[name]
+                print(
+                    f"Name: {name}, Address: {hex(cur_sym['address'])}, Size: {hex(cur_sym['size'])}, Padding: {hex(cur_sym['padding'])}"
+                )
+    
+class parser:
+    def __init__(self, symbol, inst, register, execute, exit_pc, acquire_stack_address):
+        self.symbol = symbol 
+        self.inst = inst
+        self.stacklist = []
+        self.stackfunction = []
+        self.register = register
+        self.execute = execute
+        self.edge = set()
+        self.vertex = set()
+        self.index = 0
+        self.exit_pc = exit_pc
+        self.retjmppc = 0
+        self.retjmpflag = 0
+        self.acquire_stack_address = acquire_stack_address
+        self.retcallpc = []
+        self.seenlist = [] ## handle the while loop jmp.
+        
+    def stack_analyzer(self):
+        index_list = list(self.inst.keys())
+        index_list.append(-1) ## dummy value for last iteration.
+        self.index = index_list.index(self.register.reg["pc"])
+        nextinstRip = list(self.inst.keys())
+        nextinstRip.append(-1) ## dummy value for last iteration.
+        while(self.register.reg["pc"] != self.exit_pc):
+            self.register.updaterip(nextinstRip[self.index + 1 if self.index + 1 in nextinstRip else self.index]) ## catch the rip for memory instruction.
+            if self.register.reg["pc"] in self.symbol.keys():  ## check function block (as basic block but we use function as unit.)
+                self.stackfunction.append(self.symbol[self.register.reg["pc"]])
+                logstack(self.symbol[self.register.reg["pc"]])   ## TODO: here is error.
+                self.register.updatestackreg(self.symbol[self.register.reg["pc"]] == 'custom_acquire_stack') ## if it is acquiring stack address, do not setting the stack size.
+                self.stacklist.append(self.register.reg["stack"])
+                self.register.clean()
+                
+                ###### Graph
+                vertexfrom = self.register.reg["pc"]
+                self.vertex.add(vertexfrom)
+                ######
+            self.execute.exe(self.inst[self.register.reg["pc"]], self.edge, vertexfrom)
+            
+            
+            #### set up next instruction pc
+    
+            if (self.index == index_list.index(self.register.reg["pc"])):  ## fetch next instruction
```

Comment:
This is all quite a bit more complicated than I was expecting. Great job figuring it all out!